File: Seminar/src/main/java/org/sopt/Seminar/global/common/dto/ErrorResponse.java
package org.sopt.Seminar.global.common.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.sopt.Seminar.global.common.exception.SoptException;

@Getter
@AllArgsConstructor
public class ErrorResponse {
    private int errorCode;
    private String errorType;
    private String detail;

    public ErrorResponse(SoptException soptException) {
        this.errorCode = soptException.getErrorCode();
        this.errorType = soptException.getErrorType().toString();
        this.detail = soptException.getDetail();
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/global/common/dto/ErrorResponse.kt
package org.sopt.kotlinsemina.global.common.dto

import org.sopt.kotlinsemina.global.common.exception.SoptException

data class ErrorResponse(
    val errorCode: Int,
    val errorType: String,
    val detail: String
) {
    constructor(soptException: SoptException): this(
        soptException.errorCode,
        soptException.errorType.toString(),
        soptException.detail
    )
}
